src/utility/resource_lock.cpp

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2011-2015 libbitcoin developers (see AUTHORS)
+ *
+ * This file is part of libbitcoin.
+ *
+ * libbitcoin is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License with
+ * additional permissions to the one published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version. For more information see LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <bitcoin/bitcoin/utility/resource_lock.hpp>
+
```

Comment:
By "main hpp" do you mean resource_lock.hpp? If so that's okay, but I prefer to include each reference in the file where it issued. This makes it safe to remove from a file if no longer used locally. Determine the transitive closure of all includes is humanly impossible, I got tired of chasing unpredictable compiler errors downstream of a seemingly innocuous change.